modules/correlation/id-counter.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 OneIdentity LLC.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef ID_COUNTER_H_INCLUDED
+#define ID_COUNTER_H_INCLUDED
+
+#include "glib.h"
+
+typedef struct _IdCounter
+{
+  guint id;
+  guint ref_count;
+} IdCounter;
```

Comment:
[optional]

These are technically private fields, so the struct definition should be in the `.c` file.

Reply:
Thanks for the suggestion, I have moved the struct definition to the `.c` file.